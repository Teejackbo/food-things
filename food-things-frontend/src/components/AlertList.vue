<template>
  <transition-group name="slide-fade" tag="section">
    <Alert
      v-for="(alert, index) in alerts"
      :key="index"
      :alert="alert" />
  </transition-group>
</template>

<script>
import { mapState } from 'vuex'
import Alert from './Alert.vue'

export default {
  name: 'AlertList',

  components: {
    Alert,
  },

  computed: mapState({
    alerts: state => state.alerts.alerts,
  }),
}
</script>

<style lang="scss" scoped>
@import '../style/variables';

.slide-fade-enter-active,
.slide-fade-leave-active {
  transition: all 0.5s ease;
}

.slide-fade-leave-active {
  position: absolute;
  bottom: 0;
  right: 0;
}

.slide-fade-enter,
.slide-fade-leave-to {
  transform: translateX(50px);
  opacity: 0;
}

section {
  @media (max-width: 767px) {
    bottom: 1em;
    right: 3em;
    width: 50%;
  }
  bottom: 3em;
  position: fixed;
  right: 4em;
  z-index: 100;
  width: 12%;
}
</style>
