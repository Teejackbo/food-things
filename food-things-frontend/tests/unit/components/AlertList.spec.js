import { mount } from '@vue/test-utils'
import AlertList from '@/components/AlertList.vue'

describe('AlertList', () => {
  const alerts = [
    {
      message: 'Alert 1.',
      type: 'success',
      timeout: 3000,
    },
    {
      message: 'Alert 2.',
      type: 'warning',
      timeout: 3000,
    },
  ]

  function getWrapper() {
    return mount(AlertList, {
      mocks: {
        $store: {
          state: {
            alerts: {
              alerts,
            },
          },
        },
      },
    })
  }

  it('Should match snapshot.', () => {
    const wrapper = getWrapper()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should render a transition group with the correct attributes.', () => {
    const wrapper = getWrapper()

    expect(wrapper.attributes().name).toBe('slide-fade')
    expect(wrapper.attributes().tag).toBe('section')
  })

  it('Should render the alerts from the store correctly.', () => {
    const wrapper = getWrapper()
    const children = wrapper.findAll('div')

    expect(children).toHaveLength(2)
    expect(children.at(0).classes()).toContain('success')
    expect(children.at(0).text()).toBe('Alert 1.')
    expect(children.at(1).classes()).toContain('warning')
    expect(children.at(1).text()).toBe('Alert 2.')
  })
})
