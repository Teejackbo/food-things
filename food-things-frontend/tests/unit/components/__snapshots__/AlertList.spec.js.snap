// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertList /> Should match snapshot. 1`] = `
<span
  name="slide-fade"
  tag="section"
>
  <div
    class="alert success"
  >
    
  Alert 1.

  </div>
  <div
    class="alert warning"
  >
    
  Alert 2.

  </div>
</span>
`;
